src/LoopCarry.cpp

```diff
@@ -231,17 +231,17 @@ class LoopCarryOverLoop : public IRMutator {
                 stores.push_back(v[i]);
             } else {
                 if (!stores.empty()) {
-                    result.push_back(lift_carried_values_out_of_stmt(Block::make(stores)));
+                    result.push_back(lift_carried_values_out_of_stmt(Block::make(stores, op->ordering)));
```

Comment:
I wonder if block_to_vector needs to be updated. I think there might be the case when inner block has different ordering, but it will be peeled off to the vector, and then when all the statements are put back into the block, inner block ordering will be overridden with outer block ordering.

Reply:
ugh yes, preserving it is going to be annoying